SampleGallery/Samples/SDK Insider/Gears/Gears.cs

```diff
@@ -13,126 +13,258 @@
-            _gearVisuals[0].StartAnimation("RotationAngleInDegrees", _gearMotor);
+
+            // Slow...
+            gearMotionScalarAnimation.Duration = TimeSpan.FromSeconds(5);
+
+            // I'm only animating the first gear! :-O
+            gearVisuals.First().StartAnimation("RotationAngleInDegrees", gearMotionScalarAnimation);
         }
 
+        #region Other Methods
+
         private void Stop_Click(object sender, RoutedEventArgs e)
         {
-            _gearVisuals[0].StopAnimation("RotationAngleInDegrees");
+            gearVisuals.First().StopAnimation("RotationAngleInDegrees");
         }
 
         private void Reverse_Click(object sender, RoutedEventArgs e)
         {
-            if (_gearMotor.Direction == AnimationDirection.Normal)
+            if (gearMotionScalarAnimation.Direction == AnimationDirection.Normal)
+            {
+                gearMotionScalarAnimation.Direction = AnimationDirection.Reverse;
+            }
+            else
+            {
+                gearMotionScalarAnimation.Direction = AnimationDirection.Normal;
+            }
+
+            gearVisuals.First().StartAnimation("RotationAngleInDegrees", gearMotionScalarAnimation);
+        }
+
+        private void AddXGearsButton_Click(object sender, RoutedEventArgs e)
+        {
+            var amount = int.Parse(this.NumberOfGears.Text) + this.gearVisuals.Count - 1;
+
+            this.Setup();
+            var maxAreaPerTile = Math.Sqrt((this.Container.ActualWidth * this.Container.ActualHeight) / (amount + this.Container.Children.Count));
+
+            if (maxAreaPerTile < this.width)
+            {
+                var wholeTilesHeight = Math.Floor(this.Container.ActualHeight / maxAreaPerTile);
+                var wholeTileWidth = Math.Floor(this.Container.ActualWidth / maxAreaPerTile);
+
+                this.FirstGear.Width = this.FirstGear.Height = maxAreaPerTile;
+                this.width = this.height = maxAreaPerTile;
+
+                this.x = this.gearDimension = this.width * 0.87;
+            }
+
+            for (int i = 0; i < amount; i++)
+            {
+                AddGear_Click(sender, e);
+            }
+        }
+
+        private void RotateGears_Click(object sender, RoutedEventArgs e)
+        {
+            // Slow...
+            gearMotionScalarAnimation.Duration = TimeSpan.FromSeconds(5);
+
+            var container = ElementCompositionPreview.GetElementVisual(this.Root);
+            container.AnchorPoint = new Vector2(1f, 1f);
+            //container.CenterPoint = new Vector3(0.5f, 0.5f, 0.5f);
+            // I'm only animating the first gear! :-O
+            container.StartAnimation("RotationAngleInDegrees", gearMotionScalarAnimation);
+        }
+
+        private void PerformLayoutCalculation()
+        {
+            if (((x + this.Container.Margin.Left + this.width > this.Container.ActualWidth) && gearDimension > 0) || (x < this.Container.Margin.Left && gearDimension < 0))
```

Comment:
can we reformat this condition across a few lines of code to improve readability?